parley/src/layout/alignment.rs

```diff
@@ -104,10 +120,13 @@ pub(crate) fn unjustify<B: Brush>(layout: &mut LayoutData<B>) {
             if line.break_reason != BreakReason::None && line.num_spaces != 0 {
                 let adjustment = extra / line.num_spaces as f32;
                 let mut applied = 0;
-                for line_run in layout.line_items[line.item_range.clone()]
-                    .iter()
-                    .filter(|item| item.is_text_run())
-                {
+
+                let line_items: &mut dyn Iterator<Item = &LineItemData> = if is_rtl {
+                    &mut layout.line_items[line.item_range.clone()].iter().rev()
+                } else {
+                    &mut layout.line_items[line.item_range.clone()].iter()
+                };
```

Comment:
I suspect (with a full understanding of the invariants around `num_spaces`) that it's possible to refactor this to iterate in the same order regardless of bidi (probably with a fixup to trim trailing whitespace from the slice before starting? - but I'm speculating that trailing whitespace is why `num_spaces` may not equal the actual number of space char clusters), which would be a small win for both code runtime perf and binary size.

But this seems fine for now.

Reply:
Yes, I think that's right. I also remember experimenting a bit, but it wasn't completely straightforward. Something to be revisited.